py/sight/sight.py

```diff
@@ -161,130 +160,109 @@ def _initialize_default_params(self, params: sight_pb2.params):
+    self.line_suffix = contextvars.ContextVar('line_suffix', default='')
+    self.open_block_start_locations = contextvars.ContextVar(
+        'open_block_start_locations', default=[])
+    self.num_direct_contents = contextvars.ContextVar('num_direct_contents',
+                                                      default=Location())
     self.num_transitive_contents = contextvars.ContextVar(
-        'num_transitive_contents')
-    self.num_transitive_contents.set(Location())
-    self.active_block_labels = contextvars.ContextVar('active_block_labels')
-    self.active_block_labels.set([])
-    self.active_block_start_time = contextvars.ContextVar('active_block_start_time')
-    self.active_block_start_time.set([])
-    self.active_block_deeper_elapsed_time = contextvars.ContextVar('active_block_deeper_elapsed_time')
-    self.active_block_deeper_elapsed_time.set([0])
-
+        'num_transitive_contents', default=Location())
+    self.active_block_labels = contextvars.ContextVar('active_block_labels',
+                                                      default=[])
+    self.active_block_start_time = contextvars.ContextVar(
+        'active_block_start_time', default=[])
+    self.active_block_deeper_elapsed_time = contextvars.ContextVar(
+        'active_block_deeper_elapsed_time', default=[0])
+
+  def _initialize_logging(self):
+    self.index = 1
     self.attributes = {}
     self.open = True
-
     self.id = 0
     self.set_attribute('log_uid', str(self.id))
-
-    if self.params.silent_logger:
-      return
+    # Configure the tracking state of the Sight object, which records the current location
+    # in the log of the current task, including its hierarchical nesting.
+    self.pause_logging_depth = 0
 
     # The path prefix common to all the file(s) that hold the log.
     self.path_prefix = ''
-    path_label = 'log'
+    self.path_label = 'log'
     if self.params.label:
-      path_label = self.params.label
+      self.path_label = self.params.label
 
     if self.params.local:
-      self.path_prefix = '%s/%s' % (self.params.log_dir_path, path_label)
+      self.path_prefix = '%s/%s' % (self.params.log_dir_path, self.path_label)
       self.id = 0
 
-  def __init__(
-      self,
-      params: sight_pb2.Params,
-      configuration: Optional[Sequence[sight_pb2.Object]] = None,
-  ):
-    self._initialize_default_params(params)
-
-    # Initialize each widget's state to make sure its state field is created.
-    self.widget_decision_state = defaultdict(dict)
-    self.widget_simulation_state = SimulationWidgetState()
-    # self._configure(configuration)
-
-    # Configure the tracking state of the Sight object, which records the current location
-    # in the log of the current task, including its hierarchical nesting.
-    self.pause_logging_depth = 0
-
-    self._initialize_location()
-    self._initialize_sight_params()
-
+  def _link_child_to_parent(self):
```

Comment:
these are internal but their functionality is not immediately clear from the name so please add function comments